crates/ruff_python_formatter/src/string/implicit.rs

```diff
@@ -238,26 +246,30 @@ impl<'a> FormatImplicitConcatenatedStringFlat<'a> {
                 StringLike::FString(_) => AnyStringPrefix::Format(FStringPrefix::Regular),
             };
 
-            // Only determining the preferred quote for the first string is sufficient
-            // because we don't support joining triple quoted strings with non triple quoted strings.
-            let quote = if let Ok(preferred_quote) =
-                Quote::try_from(normalizer.preferred_quote_style(first_part))
-            {
-                for part in string.parts() {
-                    let part_quote_metadata =
-                        QuoteMetadata::from_part(part, context, preferred_quote);
-
-                    if let Some(merged) = merged_quotes.as_mut() {
-                        *merged = part_quote_metadata.merge(merged)?;
-                    } else {
-                        merged_quotes = Some(part_quote_metadata);
+            let quote = if let Some(quote) = preserve_quotes_requirement {
+                quote
```

Comment:
The only change here is the addition of this condition to return the quote that needs to be preserved? I'm curious to know why was this condition not present before this? It seems like it should've been present.

Reply:
Not 100%. I would have to remind myself of how the pre 312 fstring formatting works again. It might also just be that we were missing a test.